community/kernel/src/main/java/org/neo4j/kernel/recovery/Recovery.java

```diff
@@ -82,11 +99,27 @@ public void init() throws Throwable
         }
 
         monitor.recoveryRequired( recoveryFromPosition );
+        spi.startRecovery();
+
+        // Backwards for neo store only
+        long checkpointTxId = TransactionIdStore.BASE_TX_ID;
+        try (   TransactionCursor transactionsToRecover = spi.getTransactionsInReverseOrder( recoveryFromPosition );
+                RecoveryApplier recoveryVisitor = spi.getRecoveryApplier( REVERSE_RECOVERY ); )
+        {
+            while ( transactionsToRecover.next() )
+            {
+                recoveryVisitor.visit( transactionsToRecover.get() );
+                checkpointTxId = transactionsToRecover.get().getCommitEntry().getTxId();
```

Comment:
We are only using `checkpointTxId` to report to monitor that "this is how far back I reversed the store", right? A better name for that variable would be something like `reversedToTxId`.

Reply:
Right, since it's the last checkpoint or the first transaction id ever.